Collections/script.js

```diff
@@ -256,14 +256,167 @@ console.log(filterByAge(people, 30));
+//     return userDesctiption;
+// }
+
+
+// console.log(userInfo(people))
+
 // Zadanie 12.
 
+// function olderPersons(obj){                    //funkcja wyświetlająca założony tekst na podstawie danych
+//     let overForty = [];
+//     for(let elem of obj){                      // weryfikacja wieku uzytkowników
+//         if(elem.age >= 40){
+//             overForty.push(elem)
+//         }
+//     }
+//     let textOutput = [];
+//     for(let i = 0; i < overForty.length; i++){ // pętla wprowadzająca dane użytkownika do treści
+//         if( i === 0){                          // wprowadzamy osobną treść dla pierwszego użytwkonika
+//             textOutput.push(`The user is ${overForty[i].name} and he is ${overForty[i].age} years old`)
+//         }
+//         else{                                  //wszyscy kolejni użytkoniwcy jeśli jest ich nawet więcej niż dwóch, otrzymują ten sam tekst
+//             textOutput.push(`Next user is ${overForty[i].name} and he is ${overForty[i].age} years old`)
+//         }
+//     };
+//     let sumAge = 0;
+//     for (let elem of overForty){               // pętla sumująca wiek
+//         sumAge = sumAge + elem.age
+//     }
+//     textOutput.push(`Together they have ${sumAge} years`)
+//     return textOutput;
+// }
+
+// console.log(olderPersons(people))
+
 // Zadanie 13.
 
+// function sortingByAge(obj){                          //tworzymy funckje sortującą wiek użytkowników
+//     let sorted = obj.sort((a, b) => a.age - b.age);  // korzystamy z metody sort() do posortowania użytkowników
+//     return sorted;
+// }
+
+// console.log(sortingByAge(people))
+
 // Zadanie 14.
 
+// function dayOfTheWeekYouWereBorn(str) {               // funkcja podająca dzień tygodnia z wprowadzonej daty
+//     let fullDate = new Date(str);                     // korzystamy z obiektu Date, który pzoowli nam zamienić wprowadzony string na format daty
+//     let day = fullDate.getDay();                      // korzystamy z metody getDay() która zwróci "index" dnia tygodnia
+//     let dayNames = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]   // tworzymy tabele z nazwami dni tygodnia
+
+//     return `You were born on ${dayNames[day]} `;      // wprowadzamy konkretny dzień w tekst wybierając prawidłowy dzień tygodnia zgodnie z podanym indexem
+// }
+
+// console.log(dayOfTheWeekYouWereBorn("2000-11-20"))
+
 // Zadanie 15.
 
+// function countWordOccurrences(str) {                // funkcja licząca występowanie słów
+//     let countedWords = {};  
+//     let splitedStr = str.split(" ");                // dzielimy podany string na pojedyńcze słowa
+//     for(let i = 0; i <splitedStr.length; i++){      // przechodzimy przez wszystkie słowa z tablicy
+//         let word = splitedStr[i].replace(".", "")   // tworzymy zmienną przechowująca sprawdzane słowo, jednocześnie kasujemy ewentualny znak interpunkcyjny
```

Comment:
By JavaScript i Javascript liczyło jako to samo słowo:

```suggestion
//         let word = splitedStr[i].replace(".", "").toLowerCase()   // tworzymy zmienną przechowująca sprawdzane słowo, jednocześnie kasujemy ewentualny znak interpunkcyjny
```